File: Assets/Scripts/Core/Pieces/Piece.cs
namespace Parkiller.Core
{
    public class Piece
    {
        public readonly PieceColor Color;
        public readonly int PieceIndex;

        public PieceState State = PieceState.InYard;
        public int TrackPosition = -1;
        public int CorridorPosition = -1;

        public Piece(PieceColor color, int pieceIndex)
        {
            Color = color;
            PieceIndex = pieceIndex;
        }
    }
}

File: src/core/pieces/piece.ts
import type { PieceColor } from '../pieceColor'

export type PieceState = 'InYard' | 'OnTrack' | 'InHomeCorridor' | 'Finished'

export interface Piece {
  color: PieceColor
  pieceIndex: number
  state: PieceState
  trackPosition: number
  corridorPosition: number
}

export function createPiece(color: PieceColor, pieceIndex: number): Piece {
  return {
    color,
    pieceIndex,
    state: 'InYard',
    trackPosition: -1,
    corridorPosition: -1,
  }
}
